### PR TITLE
repo-updater: Implement GitoliteSource

### PR DESCRIPTION
**Pull request by tsenart**
This change set implements the `GitoliteSource`, the first step in #3464.

@beyang: Regarding the change in ffb804d, it's meant to aid testing. I was trying to think of any security reason why we had that check but couldn't come up with anything. Can you verify this please?

Part of #3467

### CODE REVIEW

File: cmd/repo-updater/repos/sources.go [hunk 1]
```diff
@@ -74,6 +79,8 @@ func NewSource(svc *ExternalService, cf httpcli.Factory) (Source, error) {
 		return NewGitLabSource(svc, cf)
 	case "bitbucketserver":
 		return NewBitbucketServerSource(svc, cf)
+	case "gitolite":
```
Comment (beyang) [inline cmd/repo-updater/repos/sources.go]: Now that we've added this here, can we get rid of the [old Gitolite repo-updater code path](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@62b8b31401187793f1443c832782d4c7e63b4df5/-/blob/cmd/repo-updater/repos/gitolite.go#L114:6)?

Comment (tsenart) [reply cmd/repo-updater/repos/sources.go]: Not yet. This is only implementing the `Source`. There are [other things](https://github.com/sourcegraph/sourcegraph/issues/3464) we need to do before removing the old code path and before enabling the new one.

File: cmd/gitserver/server/list-gitolite.go [hunk 2]
```diff
@@ -32,38 +25,24 @@ type iGitoliteClient interface {
 // listRepos iterates through all Gitolite configs and, for each, lists the repos for the Gitolite
```
Comment (tsenart) [inline cmd/gitserver/server/list-gitolite.go]:
```suggestion
// listRepos lists the repos of a Gitolite server reachable at the address in gitoliteHost
```

File: cmd/gitserver/server/list-gitolite.go [hunk 3]
```diff
@@ -32,38 +25,24 @@ type iGitoliteClient interface {
...
 // host.
```
Comment (tsenart) [inline cmd/gitserver/server/list-gitolite.go]:
```suggestion
```

**Pull request merged**